postgres/codenamemap.yaml

```diff
@@ -40,22 +40,6 @@
 
 
 {% macro fedora_codename(name, version, codename=none) %}
-  {#
-  Generate lookup dictionary map for Fedora distributions
-
-    name:
-      distro codename
-    version:
-      PostgreSQL release version
-    codename:
-      optional grain value if `name` does not match the one returned by
-      `oscodename` grain
-  #}
-
-  {# use upstream version if configured #}
-  {% if repo.use_upstream_repo == true %}
-    {% set version = repo.version %}
-  {% endif %}
```

Comment:
You have moved `version` variable re-assignment out of `macro` scope, so it would not be altered then you have to set version for upstream packages. All macros calls will use only the version value given by the parameter, ignoring the Pillar-given value.